### PR TITLE
Filter by PTM site match

### PR DESCRIPTION
**Pull request by tonywu1999**
## Motivation and Context

This PR adds PTM-site-based filtering to INDRA-derived subnetworks to make extracted interactions more relevant for phospho/PTM-centric proteomics analyses. It refactors the INDRA filtering pipeline into discrete helper steps (PTM-site filtering then curation filtering) and exposes new visualization font-size parameters for HTML/Cytoscape outputs. The aim is to (1) allow optional removal of INDRA edges that do not match input PTM-site information and (2) separate curation adjustments into a dedicated helper.

## Solution summary

- Introduced an optional parameter to filter INDRA edges by matching PTM site information from input PTM results prior to curation filtering.
- Refactored the INDRA filtering pipeline into .filterByPtmSite() and .filterByCuration() helpers and updated getSubnetworkFromIndra to run PTM-site filtering then curation filtering, returning the chained subnetwork result.
- Removed api_key usage from several INDRA-related helpers and from the main call site.
- Added node font-size parameters to the HTML/Cytoscape visualization API and propagated the value through the config generation pipeline.

## Detailed changes

- R/getSubnetworkFromIndra.R
  - Replaced api_key parameter with filter_by_ptm_site = FALSE in the public signature.
  - Added call to .filterByPtmSite(nodes, edges, filter_by_ptm_site) and then .filterByCuration(...) — now returns the filtered subnetwork.
  - Removed passing api_key into .filterIndraResponse; .filterIndraResponse is now called without api_key.

- R/utils_getSubnetworkFromIndra.R
  - Removed api_key from .get_incorrect_curation_count() and adjusted its request logic to query INDRA curation list without requiring an API key.
  - Simplified .filterIndraResponse() signature by removing api_key and filter_by_curation; inlined curation-adjusting logic was removed.
  - Added .filterByPtmSite(nodes, edges, filter_by_ptm_site): filters edges lacking INDRA site matches to input PTM sites using calculatePTMOverlapAggregated().
  - Added .filterByCuration(nodes, edges, evidence_count_cutoff, filter_by_curation): queries incorrect curation counts and adjusts/evaluates evidence counts before filtering edges.
  - Reworked code comments and roxygen docs to reflect new helper functions and parameter changes.

- R/visualizeNetworksWithHTML.R, man/exportNetworkToHTML.Rd, man/generateCytoscapeConfig.Rd, man/previewNetworkInBrowser.Rd
  - Added nodeFontSize (public) / node_font_size (internal) parameter with default 12 to control node label font size in HTML/Cytoscape visualizations.
  - Propagated the font-size parameter through exportNetworkToHTML -> generateCytoscapeConfig -> createNodeElements.

- Documentation
  - Updated roxygen/Rd docs for getSubnetworkFromIndra to replace api_key with filter_by_ptm_site and document applicability (differential PTM abundance use case).
  - Updated man pages for visualization functions to document new font-size parameters.

- Public API changes
  - getSubnetworkFromIndra signature changed; default behavior preserved but parameter list altered (breaking change).
  - Return value now the filtered subnetwork (output of new helper chain) rather than previously assembled nodes/edges list.

## Unit tests added or modified

- No new unit tests were added for:
  - filter_by_ptm_site behavior (.filterByPtmSite)
  - .filterByCuration behavior and interaction with INDRA curation counts
  - The path combinations (filter_by_ptm_site TRUE/FALSE + filter_by_curation TRUE/FALSE)
  - Visualization parameter propagation (nodeFontSize/node_font_size)
- Existing tests for getSubnetworkFromIndra remain limited to basic structure and input validation and do not exercise the new filtering logic. The PR’s description/checklist shows testing steps left empty.

## Coding guidelines and potential issues

- Breaking API change: Removing api_key from public getSubnetworkFromIndra is a breaking change with no deprecation/migration guidance in docs or the PR description.
- Missing tests: New functionality and refactored paths lack unit tests and mocks to validate behavior and edge cases.
- Potential runtime/scoping bug: .filterByCuration() adjusts and filters edges by evidence_count_cutoff; verify that evidence_count_cutoff is passed into or available to that function (the diff summary suggests the helper receives evidence_count_cutoff but reviewers should confirm the implementation does not rely on an unbound variable).
- Behavior change risk: .filterIndraResponse() no longer adjusts evidence counts via api-key-based curation lookup; ensure the new .filterByCuration() correctly reimplements prior behavior and that removing api_key does not break authenticated endpoints or rate-limited access in deployment contexts.
- Documentation vs. implementation: The PR updates Rd/roxygen docs but checklist items for contributing guidelines, change warnings, dependent changes, code styling, and documentation generation are unchecked — ensure CI/doc build and contribution-policy compliance before merge.

### CODE REVIEW

File: R/utils_getSubnetworkFromIndra.R [hunk 1]
```diff
@@ -373,6 +360,30 @@
...
+        edges$evidenceCount <- edges$evidenceCount - incorrect_counts
+        edges <- edges[edges$evidenceCount >= evidence_count_cutoff, ]
+        nodes <- nodes[nodes$id %in% c(edges$source, edges$target), ]
+    }
+    return(list(nodes = nodes, edges = edges))
+}
+
+.filterByPtmSite = function(nodes, edges, filter_by_ptm_site) {
+    if (filter_by_ptm_site && nrow(nodes[!is.na(nodes$Site), ]) > 0) {
+        ptm_overlap <- calculatePTMOverlapAggregated(edges, nodes)
```
[the remark this answers is not in the document]
Comment (tonywu1999) [reply R/utils_getSubnetworkFromIndra.R]: Any other options?  Why do my other functions like .filterByCuration work?
